tests/Constraints/CoerciveTest.php

```diff
@@ -130,7 +130,41 @@ public function dataCoerceCases()
             'string', 'integer', 42, true
         );
 
-        $tests = array();
+        // #46 check coercion with "const"
+        $tests[] = array(
+            '{"properties":{"propertyOne":{"type":"string","const":"42"}}}',
+            '{"propertyOne":42}',
+            'integer', 'string', '42', true
+        );
+
+        // #47 check coercion with "const"
+        $tests[] = array(
+            '{"properties":{"propertyOne":{"type":"number","const":42}}}',
+            '{"propertyOne":"42"}',
+            'string', 'integer', 42, true
+        );
+
+        // #48 check boolean coercion with "const"
+        $tests[] = array(
+            '{"properties":{"propertyOne":{"type":"boolean","const":false}}}',
+            '{"propertyOne":"false"}',
+            'string', 'boolean', false, true
+        );
+
+        // #49 check boolean coercion with "const"
+        $tests[] = array(
+            '{"properties":{"propertyOne":{"type":"boolean","const":true}}}',
+            '{"propertyOne":"true"}',
+            'string', 'boolean', true, true
+        );
+
+        // #50 check boolean coercion with "const"
+        $tests[] = array(
+            '{"properties":{"propertyOne":{"type":"boolean","const":true}}}',
+            '{"propertyOne":1}',
+            'integer', 'boolean', true, true
+        );
```

Comment:
Please also add some deep object tests. With and without mismatched property types.